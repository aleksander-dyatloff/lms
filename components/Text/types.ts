namespace TextComponent {
  export type Typography = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'span' | 'p' | 'sub'

  export interface Props {
    as?: Typography
    variant?: Typography
    className?: string
  }
}

export default TextComponent
