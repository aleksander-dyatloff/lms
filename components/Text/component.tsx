import combineClasses from '@utils/combineClasses'
import { FC } from 'react'

import Wrapper from './styles'
import TextComponent from './types'

const Text: FC<TextComponent.Props> = ({
  as = 'span',
  className,
  variant = as,
  ...restProps
}) => {
  const wrapperClasses = combineClasses(variant, className)

  return (
    <Wrapper
      className={wrapperClasses}
      as={as}
      {...restProps}
    />
  )
}

export default Text
