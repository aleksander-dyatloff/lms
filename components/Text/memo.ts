import { memo } from 'react'

import Text from './component'

export default memo(Text)
