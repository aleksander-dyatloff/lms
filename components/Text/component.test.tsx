import 'jest-styled-components'

import { render } from '@testing-library/react'
import { Profiler } from 'react'

import Text from './component'

describe('Text component', () => {
  test('should match to snapshot', () => {
    const { container } = render(<Text />)

    expect(container).toMatchSnapshot()
  })

  test('first render duration should match to specification', () => {
    render(
      <Profiler
        id='Text'
        onRender={(_, __, renderDuration) => {
          expect(renderDuration).toBeLessThanOrEqual(5)
        }}
      >
        <Text />
      </Profiler>,
    )
  })
})
