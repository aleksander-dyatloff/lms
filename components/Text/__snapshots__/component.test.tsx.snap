// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should match to snapshot 1`] = `
.c0.h1 {
  font-size: 2.2rem;
  font-weight: 600;
}

.c0.h2 {
  font-size: 2rem;
  font-weight: 600;
}

.c0.h3 {
  font-size: 1.8rem;
  font-weight: 600;
}

.c0.h4 {
  font-size: 1.6rem;
  font-weight: 500;
}

.c0.h5 {
  font-size: 1.4rem;
  font-weight: 500;
}

.c0.h6 {
  font-size: 1.2rem;
  font-weight: 500;
}

.c0.p {
  line-height: 1.4;
}

.c0.sub {
  font-size: 0.8rem;
  opacity: 0.7;
}

<div>
  <span
    class="c0  span reset"
  />
</div>
`;
